components/kubeflow/kfserving/src/kfservingdeployer.py

```diff
@@ -122,8 +104,74 @@ def kfserving_deployment(metadata, default_model_spec, canary_model_spec=None, c
+    parser.add_argument('--namespace', type=str, help='Kubernetes namespace where the KFServing service is deployed.', default='kubeflow')
+    parser.add_argument('--framework', type=str, help='Model Serving Framework', default='tensorflow')
+    parser.add_argument('--default-custom-model-spec', type=json.loads, help='Custom runtime default custom model container spec', default={})
+    parser.add_argument('--canary-custom-model-spec', type=json.loads, help='Custom runtime canary custom model container spec', default={})
+    parser.add_argument('--kfserving-deployer-api', type=str, help='kfserving remote deployer api endpoint', default='')
+    parser.add_argument('--autoscaling-target', type=str, help='Autoscaling target number', default='0')
+    parser.add_argument('--output_path', type=str, help='Path to store URI output')
+    args = parser.parse_args()
+
+    url = re.compile(r"https?://")
+
+    action = args.action.lower()
+    model_name = args.model_name
+    default_model_uri = args.default_model_uri
+    canary_model_uri = args.canary_model_uri
+    canary_model_traffic = int(args.canary_model_traffic)
+    namespace = args.namespace
+    framework = args.framework.lower()
+    output_path = args.output_path
+    default_custom_model_spec = args.default_custom_model_spec
+    canary_custom_model_spec = args.canary_custom_model_spec
+    kfserving_deployer_api = url.sub('', args.kfserving_deployer_api)
+    autoscaling_target = int(args.autoscaling_target)
+
+    if kfserving_deployer_api:
+        formData = {
+            "action": action,
+            "model_name": model_name,
+            "default_model_uri": default_model_uri,
+            "canary_model_uri": canary_model_uri,
+            "canary_model_traffic": canary_model_traffic,
+            "namespace": namespace,
+            "framework": framework,
+            "default_custom_model_spec": default_custom_model_spec,
+            "canary_custom_model_spec": canary_custom_model_spec,
+            "autoscaling_target": autoscaling_target
+            }
+        response = requests.post("http://" + kfserving_deployer_api + "/deploy-model", json=formData)
+        model_status = response.json()
+    else:
+        model_status = deploy_model(
+            action=action,
+            model_name=model_name,
+            default_model_uri=default_model_uri,
+            canary_model_uri=canary_model_uri,
+            canary_model_traffic=canary_model_traffic,
+            namespace=namespace,
+            framework=framework,
+            default_custom_model_spec=default_custom_model_spec,
+            canary_custom_model_spec=canary_custom_model_spec,
+            autoscaling_target=autoscaling_target
+        )
+    print(model_status)
+    try:
+        print(model_status['status']['url'] + ' is the knative domain header. $CLUSTER_IP:31380 is the endpoint to post predictions to the model.')
```

Comment:
why 31830 is hardcoded?

Reply:
The default istio ingress gateway is hosted with nodePort 31380. I updated this section with some bash command to get the LoadBalancer IP to make it more clarify.